backend/src/v1/data/repositories/wordRepository.ts

```diff
@@ -1,36 +1,80 @@
 import { Word } from "../models/models";
-
-interface WordRepository {
-    getWords(filterParams: {}): Word[];
-    getWord(id: string): Word;
-    addWord(data: AddWordData): void;
-    updateWord(word: Word): void;
-    deleteWord(id: string): void;
+import { db } from "../../../db/database";
+import { DeleteResult, InsertResult } from "kysely";
+export interface WordRepository {
+    getWords(languageCode?: string): Promise<Word[]>;
+    getWord(id: number): Promise<Word>;
+    insertWord(data: WordInsert): Promise<{ insertedRows: string; }>;
+    updateWord(data: WordUpdate): Promise<{ updatedRows: string; }>;
+    deleteWord(id: number): Promise<{ deletedRowsCount: string; }>;
 };
 
 const WordRepositoryImpl: WordRepository = {
```

Comment:
Any reason to put all these in an object, as opposed to making them top-level functions in this module?

```ts
export async function getWords(languageCode: string): Promise<Word[]> {
  // ...
}

// ...
```